src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/PooledTxsRequestor.cs

```diff
@@ -30,78 +30,64 @@ public PooledTxsRequestor(ITxPool txPool, ITxPoolConfig txPoolConfig)
-                    }
-
-                    if (hashesToRequest.Count > 0)
+                    if (hashesToRequest.Count % MaxNumberOfTxsInOneMsg == 0)
                     {
                         RequestPooledTransactionsEth66(send, hashesToRequest);
+                        hashesToRequest = new(MaxNumberOfTxsInOneMsg);
                     }
+
+                    hashesToRequest.Add(discoveredTxHashes[i]);
                 }
+
+                RequestPooledTransactionsEth66(send, hashesToRequest);
             }
         }
 
         public void RequestTransactionsEth68(Action<V66.Messages.GetPooledTransactionsMessage> send, IReadOnlyList<Hash256> hashes, IReadOnlyList<int> sizes, IReadOnlyList<byte> types)
         {
             using ArrayPoolList<(Hash256 Hash, byte Type, int Size)> discoveredTxHashesAndSizes = AddMarkUnknownHashesEth68(hashes, sizes, types);
+            if (discoveredTxHashesAndSizes.Count == 0) return;
+
+            int packetSizeLeft = TransactionsMessage.MaxPacketSize;
+            ArrayPoolList<Hash256> hashesToRequest = new(discoveredTxHashesAndSizes.Count);
 
-            if (discoveredTxHashesAndSizes.Count != 0)
+            for (int i = 0; i < discoveredTxHashesAndSizes.Count; i++)
             {
-                int packetSizeLeft = TransactionsMessage.MaxPacketSize;
-                ArrayPoolList<Hash256> hashesToRequest = new(discoveredTxHashesAndSizes.Count);
+                int txSize = discoveredTxHashesAndSizes[i].Size;
+                TxType txType = (TxType)discoveredTxHashesAndSizes[i].Type;
 
-                for (int i = 0; i < discoveredTxHashesAndSizes.Count; i++)
+                if (txSize > packetSizeLeft && hashesToRequest.Count > 0)
                 {
-                    int txSize = discoveredTxHashesAndSizes[i].Size;
-                    TxType txType = (TxType)discoveredTxHashesAndSizes[i].Type;
-
-                    if (txSize > packetSizeLeft && hashesToRequest.Count > 0)
-                    {
-                        RequestPooledTransactionsEth66(send, hashesToRequest);
-                        hashesToRequest = new ArrayPoolList<Hash256>(discoveredTxHashesAndSizes.Count);
-                        packetSizeLeft = TransactionsMessage.MaxPacketSize;
-                    }
-
-                    if (_txPoolConfig.BlobsSupport.IsEnabled() || txType != TxType.Blob)
-                    {
-                        hashesToRequest.Add(discoveredTxHashesAndSizes[i].Hash);
-                        packetSizeLeft -= txSize;
-                    }
+                    RequestPooledTransactionsEth66(send, hashesToRequest);
+                    hashesToRequest = new ArrayPoolList<Hash256>(discoveredTxHashesAndSizes.Count);
+                    packetSizeLeft = TransactionsMessage.MaxPacketSize;
                 }
 
-                if (hashesToRequest.Count > 0)
+                if (_txPoolConfig.BlobsSupport.IsEnabled() || txType != TxType.Blob)
```

Comment:
`_txPoolConfig.BlobsSupport.IsEnabled()` should be captured to readonly field of the class and not pinged every time.